bootcfg/http/generic.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/coreos/coreos-baremetal/bootcfg/server"
+	pb "github.com/coreos/coreos-baremetal/bootcfg/server/serverpb"
+)
+
+// genericHandler returns a handler that responds with generic file for
+// the requester.
+func genericHandler(srv server.Server) ContextHandler {
+	fn := func(ctx context.Context, w http.ResponseWriter, req *http.Request) {
+		group, err := groupFromContext(ctx)
+		if err != nil || group.Profile == "" {
+			http.NotFound(w, req)
+			return
+		}
+		profile, err := srv.ProfileGet(ctx, &pb.ProfileGetRequest{Id: group.Profile})
+		if err != nil || profile.GenericId == "" {
+			http.NotFound(w, req)
+			return
+		}
+		contents, err := srv.GenericGet(ctx, profile.GenericId)
+		if err != nil {
+			http.NotFound(w, req)
+			return
+		}
+
+		// collect data for rendering
+		data := make(map[string]interface{})
+		if group.Metadata != nil {
+			err = json.Unmarshal(group.Metadata, &data)
+			if err != nil {
+				log.Errorf("error unmarshalling metadata: %v", err)
+				http.NotFound(w, req)
+				return
+			}
+		}
```

Comment:
You can add the same special value `data["query"] = req.URL.RawQuery` that is provided to Ignition templates. Its acceptable here because people will use generic for experimentation and edge cases.